extension_service/datastore/providers/firestore.py

```diff
@@ -0,0 +1,281 @@
+                        "departure_gate": flight.departure_gate,
+                        "arrival_gate": flight.arrival_gate,
+                    }
+                )
+            )
+            if len(create_flights_tasks) % 10000 == 0:
+                # avoid gRPC batch write timeout error
+                await asyncio.gather(*create_flights_tasks)
+                create_flights_tasks.clear()
+        await asyncio.gather(*create_flights_tasks)
+
+    async def export_data(
+        self,
+    ) -> tuple[list[models.Airport], list[models.Amenity], list[models.Flight]]:
+        airport_docs = self.__client.collection("airports").stream()
+        amenities_docs = self.__client.collection("amenities").stream()
+        flights_docs = self.__client.collection("flights").stream()
+
+        airports = []
+
+        async for doc in airport_docs:
+            airport_dict = doc.to_dict()
+            airport_dict["id"] = doc.id
+            airports.append(models.Airport.model_validate(airport_dict))
+
+        amenities = []
+        async for doc in amenities_docs:
+            amenity_dict = doc.to_dict()
+            amenity_dict["id"] = doc.id
+            amenities.append(models.Amenity.model_validate(amenity_dict))
+
+        flights = []
+        async for doc in flights_docs:
+            flight_dict = doc.to_dict()
+            flight_dict["id"] = doc.id
+            flights.append(models.Flight.model_validate(flight_dict))
+
+        return airports, amenities, flights
+
+    async def get_airport_by_id(self, id: int) -> Optional[models.Airport]:
+        query = self.__client.collection("airports").where(
+            filter=FieldFilter("id", "==", id)
+        )
+        return models.Airport.model_validate(query.stream().to_dict())
+
+    async def get_airport_by_iata(self, iata: str) -> Optional[models.Airport]:
+        query = self.__client.collection("airports").where(
+            filter=FieldFilter("iata", "==", iata)
+        )
+        return models.Airport.model_validate(query.stream().to_dict())
+
+    async def search_airports(
+        self,
+        country: Optional[str] = None,
+        city: Optional[str] = None,
+        name: Optional[str] = None,
+    ) -> list[models.Airport]:
+        query = self.__client.collection("airports")
+
+        if country is not None:
+            query = query.where("country", "==", country)
+
+        if city is not None:
+            query = query.where("city", "==", city)
+
+        if name is not None:
+            query = query.where("name", ">=", name).where("name", "<=", name + "\uf8ff")
+
+        docs = query.stream()
+        airports = []
+        async for doc in docs:
+            airports.append(models.Airport.model_validate(dict(doc)))
+        return airports
+
+    async def get_amenity(self, id: int) -> Optional[models.Amenity]:
+        query = (
+            self.__client.collection("amenities")
+            .where(filter=FieldFilter("id", "==", id))
```

Comment:
What the difference between using FieldFilter vs where directly>

Reply:
Firestore doesn't mention any difference on their official documentation... But it does seem like most of their filter examples use `FieldFilter` instead of passing in directly. I guess it might be mostly for params passing purposes (so you could pass other params into the `where` method besides the filter).